src/repository.c

```diff
@@ -812,8 +812,13 @@ int git_repository_open_ext(
 	error = find_repo(
 		&gitdir, &workdir, &gitlink, &commondir, start_path, flags, ceiling_dirs);
 
-	if (error < 0 || !repo_ptr)
+	if (error < 0 || !repo_ptr) {
```

Comment:
Instead of adding duplicate `free`s here, I would encourage you to jump to the cleanup label so that we can consolidate cleanup in a single place.  That's the general pattern we try to use, it makes it less likely that we would end up freeing in one return path but not the other if we end up adding new allocations (for instance).

Reply:
That makes sense, I've updated the review. I initialised 'repo' to NULL in order to handle the cleanup case where error < 0 and repo_ptr is NULL.